### PR TITLE
Add docs for new Union Relationship Types Scan

### PR DESCRIPTION
**Pull request by pontusmelke**
`DirectedUnionRelationshipTypeScan`
`UnDirectedUnionRelationshipTypeScan`

### CODE REVIEW

**Comment by davidoliverSP2** [comment]
@pontusmelke Please could you try rebasing this PR? The recently merged #1511 should give us green builds. Many thanks!

**Comment by pontusmelke** [comment]
@davidoliverSP2 looks like it is building now

**Comment by davidoliverSP2** [comment]
@pontusmelke I think I've found the missing part to this PR - we need to add includes to the new sections in [cypher/cypher-docs/src/docs/dev/execution-plan-groups/operators.asciidoc](https://github.com/neo4j/neo4j-documentation/blob/dev/cypher/cypher-docs/src/docs/dev/execution-plan-groups/operators.asciidoc)

I think the syntax we need to add is:
```
// directedUnionRelationshipTypeScan
include::../ql/query-plan/directed-union-relationship-types-scan.asciidoc[]
// undirectedUnionRelationshipTypeScan
include::../ql/query-plan/undirected-union-relationship-types-scan.asciidoc[]
```

I'm not sure where in that list we want this to appear though.

**Review by davidoliverSP2 — APPROVED** [review]
@pontusmelke This is all looking good now
Screenshot 2022-07-21 at 16 04 22

File: cypher/cypher-docs/src/docs/dev/execution-plans-operator-summary.asciidoc [hunk 1]
```diff
@@ -651,6 +651,18 @@ Tests for the absence of a pattern predicate if an expression predicate evaluate
...
+|
+|
+|
+
+| <<query-plan-undirected-union-relationship-types-scan,Undirected Union Relationship Types Scan>>
+| Fetches all relationships and their start and end nodes with at least one of the provided types from the relationship type index.
+|
+|
+|
+
```
Comment (davidoliverSP2) [inline cypher/cypher-docs/src/docs/dev/execution-plans-operator-summary.asciidoc:665]: @pontusmelke I've added these operators to the ["at a glance" page](https://development.neo4j.dev/docs/cypher-manual/5.0/execution-plans/operator-summary/)

When this is building, I'll grab a screenshot for how it renders. If any entries are required in the `Leaf?`,  `Updating?` or `Condiderations` columns, please let me know.

Comment (davidoliverSP2) [reply cypher/cypher-docs/src/docs/dev/execution-plans-operator-summary.asciidoc:665]: Screenshot 2022-07-22 at 10 25 28

**Comment by davidoliverSP2** [comment]
FYI @martin-neotech

File: cypher/cypher-docs/src/docs/dev/execution-plans-operator-summary.asciidoc [hunk 2]
```diff
@@ -651,6 +651,18 @@ Tests for the absence of a pattern predicate if an expression predicate evaluate
 |
 |
 
+| <<query-plan-directed-union-relationship-types-scan,Directed Union Relationship Types Scan>>
+| Fetches all relationships and their start and end nodes with at least one of the provided types from the relationship type index.
+|
+|
+|
+
+| <<query-plan-undirected-union-relationship-types-scan,Undirected Union Relationship Types Scan>>
```
Comment (martin-neotech) [inline cypher/cypher-docs/src/docs/dev/execution-plans-operator-summary.asciidoc:660]: `<<query-plan-undirected-union-relationship-types-scan,UndirectedUnionRelationshipTypesScan>>`

File: cypher/cypher-docs/src/docs/dev/execution-plans-operator-summary.asciidoc [hunk 3]
```diff
@@ -651,6 +651,18 @@ Tests for the absence of a pattern predicate if an expression predicate evaluate
 |
 |
 
+| <<query-plan-directed-union-relationship-types-scan,Directed Union Relationship Types Scan>>
```
Comment (martin-neotech) [inline cypher/cypher-docs/src/docs/dev/execution-plans-operator-summary.asciidoc:654]: `<<query-plan-directed-union-relationship-types-scan,DirectedUnionRelationshipTypesScan>>`

Comment (martin-neotech) [reply cypher/cypher-docs/src/docs/dev/execution-plans-operator-summary.asciidoc:654]: Also this is placed on the wrong place, this page has the operators in sorted order.

**Pull request merged**